### ISSUE TITLE
`ilab generate` fails when automatically connecting to server hosting model

### ISSUE THREAD

**Issue by hickeyma** [instructlab/instructlab#1062]
When you run `ilab generate` without first starting the server hosting model (i.e. want the command to start the server automatically), you get the following error:

```command
$ ilab generate

Generating synthetic data using 'merlinite-7b-lab-Q4_K_M' model, taxonomy:'taxonomy' against http://127.0.0.1:52932/v1 server
INFO 2024-05-01 12:23:15,620 rouge_scorer.py:83 Using default tokenizer.
  0%|                                                                                                        | 0/100 [00:00<?, ?it/s]Cannot find prompt.txt. Using default prompt depending on model-family.
Synthesizing new instructions. If you aren't satisfied with the generated instructions, interrupt training (Ctrl-C) and try adjusting your YAML files. Adding more examples may help.
INFO 2024-05-01 12:23:15,621 generate_data.py:468 Selected taxonomy path compositional_skills->writing->freeform->jokes->puns-copy->general
Generating dataset failed with the following error: There was a problem connecting to the server Connection error.
  0%|                                                                                                        | 0/100 [00:02<?, ?it/s]
```
Tested with the latest on `main` branch.

**Comment by leseb** [comment]
I cannot repro this, is it failing 100% of the time?

**Comment by russellb** [comment]
I was able to reproduce this failure on mac, but once I switched to the branch for #1050, it works.  It'd be nice to see if that fixes it for anyone else that can reproduce it.

**Comment by cdoern** [comment]
@hickeyma can you reproduce anymore?

on my machine, I can no longer reproduce when building from main.

**Comment by russellb** [comment]
We have two successful test reports on systems that failed before. I'm going to close this as fixed by #1050 but if anyone reproduces, please reopen!
